pkg/schema/schema.go

```diff
@@ -5,23 +5,43 @@ package schema
 
 import (
 	"fmt"
-
 	"github.com/k14s/ytt/pkg/template"
 	"github.com/k14s/ytt/pkg/yamlmeta"
 )
 
 type AnySchema struct {
 }
 
+type NullSchema struct {
```

Comment:
As far as I can tell, the only distinction between AnySchema and NullSchema is in `ValidateWithValues`. Are there other benefits to creating this struct?

Reply:
Yes that is the only distinction between AnySchema and NullSchema. My understanding is that we are using polymorphism here to communicate different behavior for when a schema isn't provided vs when a schema is completely permissive. John has added this as a tech forum topic to discuss more as a team.